lib/fathead/pep/parse.py

```diff
@@ -0,0 +1,233 @@
+        """
+        Gets the contents from the HTML file.
+        """
+        self.url = __PYBASE__ + link['href']
+        html = request.urlopen(self.url)
+        self.set_pep_document(html.read().decode("utf-8"))
+
+    def set_pep_document(self, doc):
+        """
+        Makes up the HTML document.
+        """
+        self.document = BeautifulSoup(doc, "html.parser")
+
+    def set_pep_title(self):
+        """
+        Gets the title for the PEP
+        """
+        # self.title = "PEP " + self.pep
+        peptitle = self.document.find("h1", { "class": "page-title" }).get_text()
+        self.title = re.sub("--", "-", peptitle)
+
+    def set_pep_number(self):
+        """
+        Sets the PEP number. For example, if the PEP is identified as PEP 0008,
+        it will be stored as 8. If the PEP is identified as PEP 0143 it will be
+        stored as 143.
+        """
+        pep = re.findall(r'\d+', self.url)[0]
+        self.pep = pep.lstrip("0")
+
+    def set_pep_body(self):
+        """
+        Parses the raw document, cleans the text
+        """
+        try:
+            html = self.document.find("div", { "id": [
+                    "abstract",
+                    "introduction",
+                    "rationale",
+                    "motivation",
+                    "what-is-a-pep",
+                    "overview",
+                    "improving-python-zip-application-support",
+                    "scope",
+                    "abstract-and-rationale",
+                    "rationale-and-goals",
+                    "specification"
+                ]
+            })
+
+            if html is None: raise AttributeError
+
+            html = FHTEMPLATE.format(information=str(html))
+            html = html.replace("<pre class=\"literal-block\">", "<pre><code>")
+            html = html.replace("</pre>", "</code></pre>")
+            html = html.replace("<tt class=\"docutils literal\">", "<code class=\"inline\">")
+            html = html.replace("</tt>", "</code>")
+            html = re.sub(re.compile("<h1>((.|\n)+?)</h1>"), "", html)
+            html = re.sub(re.compile("<a (.+?)>"), "", html)
+            html = re.sub(re.compile("</a>"), "", html)
+            html = re.sub(re.compile("\[\d+\]"), "", html) # removes the vancouver type referencing
+            html = re.sub(re.compile("\n"), " ", html)
+            html = html.rstrip().replace("\t", "    ")
+            self.body = html
+        except AttributeError:
+            pass
+        except:
+            print("Parse Error: Investigate " + self.pep)
+
+    def set_structure(self):
+        """
+        Sets the structure for the output.txt file
+        """
+        if int(self.pep) not in [205, 210, 308]: # throw out redundent peps
+            entry = OUTPUT_TEMPLATE.format(
+                title=self.title,
+                entry_type="A",
+                redirect_title="",
+                empty_field="",
+                categories=self.pep,
```

Comment:
we don't want to categories for this Fathead. Categories are intended to group related articles. This creates a category page, that shows a carousel of tiles for all the articles in the category.

An example is `css pseudo elements`. Several CSS articles are a for a "pseudo element" so each of those are given the category "css pseudo element". That enables "css pseudo elements" to display a list of all those articles: https://duckduckgo.com/?q=css+pseudo+elements&ia=list

It also provides a link to the list of elements in the InfoBox for each of the articles that are part of that category.